### DISCUSSION TITLE
GraphQL query to return devices with a matching interface ip_address

### DISCUSSION THREAD

**Discussion by xeb10154** [nautobot/nautobot#3257] [Q&A]
Hi,

I am trying to write a GraphQL query to return a list of all devices with a matching interface IP address. This is filtered down to a specific site.

The first query below returns a list of ALL devices in a specific site with interfaces and IP address of that interface.

![image](https://user-images.githubusercontent.com/25885192/217389074-a9ef9ffd-16bf-4c3e-ada3-285cf34001b2.png)

The second query below is my attempt to modify the query to only return devices that have a matching interface IP address. All devices with no interfaces should be excluded.

![image](https://user-images.githubusercontent.com/25885192/217389205-f8d5d93f-5ce5-4046-8ac2-af384708fb88.png)

This returns an error:
`"message": "Unknown argument \"address\" on field \"ip_addresses\" of type \"InterfaceType\"."` 

According to the official docs, the IP address has an argument for `address`, so I'm not sure why this is complaining. And the argument **$interface_ip_addresses** is a list of IPs.

Appreciate any help I can get as I am new to GraphQL.

Thanks.

**Comment by jvanderaa** [comment]
How about something like this:

```
query {
  ip_addresses(address: "10.11.128.3/32") {
    id
    primary_ip4_for {
      id
      name
    }
  }
}
```

Change it around of what you are looking for.

**Reply by jvanderaa** [reply]
This was run against the Demo instance of Nautobot to get the information that would probably work.

Thoughts @xeb10154

**Reply by xeb10154** [reply]
Hi @jvanderaa ,

Thanks for your answer. I have tried this at first and it does brings back **ALL** matching IPs. The key issue with this however, is I have multiple Sites in Nautobot that also use the same host IPs. Which is why I need to filter by Site as well as filter by device interface IPs.

So far, my only solution is to use the first GraphQL query to bring back all the IPs in a Site and then filter the IP using a `for` loop in python. Takes around 5-6 seconds to run.
